src/report.ts

```diff
@@ -30,27 +34,55 @@ async function pollReport(
     throw new Error(`"project-name" must not contain spaces or special characters except [/-_]`);
   }
 
+  return initAndPollReportGeneric(initReport, getReport, options);
+}
+
+async function initAndPollScmReport(
+  options: ScmUploadReportOptions,
+): Promise<Result<AnalysisFailedResponse | ReportResult, GetAnalysisErrorCodes>> {
+  const projectId = options.projectId?.trim();
+  if (!projectId || projectId.length === 0) {
+    throw new Error('"project-id" must be provided for "report"');
+  }
+  if (!uuid.validate(projectId)) {
+    throw new Error('"project-id" must be a valid UUID');
+  }
+
+  const commitId = options.commitId?.trim();
+  if (!commitId || commitId.length === 0) {
+    throw new Error('"commit-id" must be provided for "report"');
+  }
+
+  return initAndPollReportGeneric(initScmReport, getScmReport, options);
+}
+
+async function initAndPollReportGeneric(
+  initReportFunc: typeof initReport | typeof initScmReport,
+  getReportFunc: typeof getReport | typeof getScmReport,
+  options: UploadReportOptions | ScmUploadReportOptions,
+) {
   emitter.analyseProgress({
     status: AnalysisStatus.waiting,
     progress: 0,
   });
 
   // First init the report
-  const initResponse = await initReport(options);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const initResponse = await initReportFunc(options as unknown as any);
```

Comment:
q: would this work? If so, we wouldn't need the eslint disable above 🤔 

```suggestion
  const initResponse = await initReportFunc(options as (UploadReportOptions & ScmUploadReportOptions));
```

Reply:
It would (kind of) work, but would still be casting, and to the wrong type (as it's a union, not an intersection).
Will follow @michelkaporin's suggestion, as that makes the types clearer.